packages/sanity/src/presentation/preview/IFrame.tsx

```diff
@@ -1,7 +1,75 @@
 import {Box} from '@sanity/ui'
 import {motion, type VariantLabels, type Variants} from 'framer-motion'
-import {forwardRef, type ReactEventHandler, useId} from 'react'
-import {styled} from 'styled-components'
+import {forwardRef, type ReactEventHandler, useEffect, useImperativeHandle, useRef} from 'react'
+import {createGlobalStyle, styled} from 'styled-components'
+
+import {useId} from '../useId'
+
+interface IFrameProps {
+  animate: VariantLabels
+  initial: VariantLabels
+  onLoad: ReactEventHandler<HTMLIFrameElement>
+  preventClick: boolean
+  src: string
+  variants: Variants
+  style: React.CSSProperties
+}
+
+export const IFrame = forwardRef<HTMLIFrameElement, IFrameProps>(
+  function IFrame(props, forwardedRef) {
+    const {animate, initial, onLoad, preventClick, src, variants, style} = props
+
+    const ref = useRef<HTMLIFrameElement | null>(null)
+    // Forward the iframe ref to the parent component
+    useImperativeHandle<HTMLIFrameElement | null, HTMLIFrameElement | null>(
+      forwardedRef,
+      () => ref.current,
+    )
+
+    /**
+     * Ensure that clicking outside of menus and dialogs will close as focus shifts to the iframe
+     */
+
+    useEffect(() => {
+      if (!ref.current) {
+        return undefined
+      }
+      const instance = ref.current
+      function handleBlur() {
+        if (instance !== document.activeElement) {
+          return
+        }
+
+        instance.dispatchEvent(new MouseEvent('mousedown', {bubbles: true, cancelable: true}))
+      }
+      window.addEventListener('blur', handleBlur)
+      return () => {
+        window.removeEventListener('blur', handleBlur)
+      }
+    }, [])
```

Comment:
Might end up being the same thing, but couldn't we use the useClickOutside (or something along those lines) from sanity ui?